src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs

```diff
@@ -176,23 +176,15 @@ private void RequestRefresh(AuthenticationTicket ticket)
 
         private CookieOptions BuildCookieOptions()
         {
-            var cookieOptions = new CookieOptions
-            {
-                Domain = Options.CookieDomain,
-                SameSite = Options.CookieSameSite,
-                HttpOnly = Options.CookieHttpOnly,
-                Path = Options.CookiePath ?? (OriginalPathBase.HasValue ? OriginalPathBase.ToString() : "/"),
-            };
+            var cookieOptions = Options.Cookie.Build(Context);
+            // ignore the 'Expires' value as this will be computed elsewhere
+            cookieOptions.Expires = null;
 
-            if (Options.CookieSecure == CookieSecurePolicy.SameAsRequest)
+            if (Options.Cookie.Path == null && OriginalPathBase.HasValue)
```

Comment:
The diff github is showing is a little misleading. The behavior your asking about moved into the CookieBuilder:

See https://github.com/aspnet/HttpAbstractions/blob/199b0fa212aa03a011f3c18575f0ab5102527c1a/src/Microsoft.AspNetCore.Http.Abstractions/CookieBuilder.cs#L97